### DISCUSSION TITLE
Auto Ducking

### DISCUSSION THREAD

**Discussion by Laurent0722** [missionpinball/missionpinball#106] [Ideas]
Hi,
I'm a long time reader but first post here :)

I'm currently fine tuning my homebrew machine and i'm strugling with audio ducking...
Is threre a way to set a TRACK so it automatically duck another one ?
ie: voices and/or sfx track automatically duck music track...
I've read that we must specifie ducking parametters for each individual sound but what a job to do... !

An auto ducking feature would be so much more efficient ! don't you think ?

is there a method to achieve this kind of feature ?

Thank you for your answer.

Laurent (From France)

**Comment by qcapen** [comment]
There is currently no way to achieve an auto-ducking effect in the MPF-MC audio engine. I think this is a great idea for a feature suggestion. 

In audio processing terms, auto-ducking is basically sidechain compression where the signal of one audio track is used to attenuate the signal of another track ([Dynamic Range Compression](https://en.wikipedia.org/wiki/Dynamic_range_compression), ). Unlike automatic ducking in audio/video editors which can analyze prerecorded audio signals and apply the effect appropriately, the MPF-MC audio engine uses real-time processing to generate its audio signals. Since sounds are generated by an event-driven system, real-time signal processing must be used to analyze and generate an automatic ducking effect. This kind of signal processing is not trivial to implement, meaning this is not a simple feature to add to the MPF-MC and would certainly add some processing overhead.

I will add this as a potential new feature issue to MPF-MC as it has a lot of potential to be very useful.

**Reply by Laurent0722** [reply]
Thanks for the answer
I hope one day this feature will be implemented... 'till then, let's copy and paste ducking parameters  !!!    ;-)

Laurent
